### ISSUE TITLE
Make call builder with_arg work with IDLValue

### ISSUE THREAD

**Issue by ericswanson-dfinity** [dfinity/agent-rs#132]
I recently had to add `SyncCallBuilder.with_value_arg(IDLValue)`, because `with_arg` panics:

- `IDLValue::ty()` calls unreachable() : https://github.com/dfinity/candid/blob/master/rust/candid/src/parser/value.rs#L326
- called from here: https://github.com/dfinity/candid/blob/master/rust/candid/src/ser.rs#L28

It would be nicer if `with_arg` worked with IDLValue rather than panicking.

**Comment by chenyan-dfinity** [comment]
Yes, this is intentional. Receiving `IDLValue` means it's untyped, thus needing a different API. I'm planning to change `IDLValue::ty()` to return `Type::Unknown`, but it won't help your use case. One has to look at the value of `IDLValue` to know the type.

**Comment by chenyan-dfinity** [comment]
You can now decode function reference to `candid::Func`, so `IDLValue` is no longer needed.
